test/e2e/storage/testsuites/testdriver.go

```diff
@@ -31,7 +31,7 @@ type TestDriver interface {
 	// CreateDriver creates all driver resources that is required for TestDriver method
 	// except CreateVolume. May be called more than once and should only do something on
 	// the first call.
-	CreateDriver()
+	CreateDriver(config *TestConfig)
```

Comment:
I think it would be clearer to have CreateDriver return a config and not take a pointer, instead of modifying the pointer argument. Init*Driver returns a static DriverInfo so CreateDriver should similarly return a dynamic TestConfig.

Reply:
since most of the implementations are using config.framework a bunch maybe pass framework and prefix instead and let CreateDriver construct the TestConfig in full

Reply:
It's true that the framework pointer could also be passed in earlier. However, it is unusable at init time because the framework initializes the content in a `BeforeEach`. I think it is cleaner to only grant access to the pointer when it is really initialized and needed, and not sooner.

Reply:
I ended up letting `CreateDriver` choose the prefix. This is a slight change in behavior, but once one gets used to it, it might even be better because we get different prefixes into the test output.